api/plLocks.py

```diff
@@ -0,0 +1,41 @@
+import threading
+
+
+class layerLock:
+    def __init__(self, name='init', parent=None):
+        self.name = name
+        self.lock = threading.Lock()
+        self.children = {}
+        self.parent = parent
+
+    def acquire(self):
+        self.lock.acquire()
+        for child in self.children.values():
+            child.acquire()
+
+    def release(self):
+        self.lock.release()
+        for child in self.children.values():
+            child.release()
```

Comment:
is this logic repeated from the previous locking code above? please use berkeley db instead.

Reply:
Non Factor with DB Implementation